superset-frontend/src/components/Chart/DrillDetailModal.test.tsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { render, screen, waitFor } from 'spec/helpers/testing-library';
+import { getMockStoreWithNativeFilters } from 'spec/fixtures/mockStore';
+import chartQueries, { sliceId } from 'spec/fixtures/mockChartQueries';
+import { QueryFormData } from '@superset-ui/core';
+import fetchMock from 'fetch-mock';
+import userEvent from '@testing-library/user-event';
+import DrillDetailModal from './DrillDetailModal';
+
+const chart = chartQueries[sliceId];
+const setup = (overrides: Record<string, any> = {}) => {
+  const store = getMockStoreWithNativeFilters();
+  const props = {
+    chartId: sliceId,
+    initialFilters: [],
+    formData: chart.form_data as unknown as QueryFormData,
+    ...overrides,
+  };
+  return render(<DrillDetailModal {...props} />, {
+    useRedux: true,
+    useRouter: true,
+    store,
+  });
+};
+const waitForRender = (overrides: Record<string, any> = {}) =>
+  waitFor(() => setup(overrides));
```

Comment:
Is it really needed? Don't we always wait for render by default?

Reply:
@kgabryje in this component there are several async events that would fire warnings in the console if we don't wait for the component to be fully loaded. Hence the reason for this.

Reply:
Oh that's interesting. Does it mean that we should use this pattern in other components too? Maybe this should be a helper function in `spec/helpers`?

Reply:
@kgabryje not sure if it is worth creating a function that takes a component and wraps it in a waitFor while also taking eventual overrides but if you think it can help I am happy to create it in a separate PR.

Reply:
In general, I think there is a lot of reusable helpers that we can create for this RTL tests. This could be an interesting project in order to enforce/suggest certain standards in writing RTL tests. CC @rusackas

Reply:
You can also use `findBy` in many of these tests if you want to wait for the async operations. [findBy](https://testing-library.com/docs/dom-testing-library/api-async/#findby-queries) is essentially the combination of `getBy` and `waitFor`. So this:

```
test('should render the title', async () => {
  await waitForRender();
  expect(
    screen.getByText(`Drill to detail: ${chart.form_data.slice_name}`),
  ).toBeInTheDocument();
});
```

Could be written as:

```
test('should render the title', async () => {
  setup();
  expect(
    await screen.findByText(`Drill to detail: ${chart.form_data.slice_name}`),
  ).toBeInTheDocument();
});
```

Either way, the important thing is to account for the async nature of the component as you did.